scenario/listboxselect.go

```diff
@@ -0,0 +1,178 @@
+	uplink := sessionState.Connection.Sense()
+	objectID := sessionState.IDMap.Get(settings.ID)
+	obj, err := uplink.Objects.GetObjectByID(objectID)
+	if err != nil {
+		actionState.AddErrors(errors.Wrapf(err, "Failed getting object<%s> from object list", objectID))
+		return
+	}
+
+	switch t := obj.EnigmaObject.(type) {
+	case *enigma.GenericObject:
+		genericObj := obj.EnigmaObject.(*enigma.GenericObject)
+		settings.doSelect(sessionState, actionState, genericObj)
+	default:
+		actionState.AddErrors(errors.Errorf("Expected generic object , got object type<%T>", t))
+		return
+	}
+
+	sessionState.Wait(actionState)
+}
+
+func (settings ListBoxSelectSettings) doSelect(sessionState *session.State, actionState *action.State, genericObj *enigma.GenericObject) {
+	objInstance := sessionState.GetObjectHandlerInstance(genericObj.GenericId, genericObj.GenericType)
+
+	selectPath, selectType, dataDefType, err := objInstance.GetObjectDefinition(genericObj.GenericType)
+	if err != nil {
+		actionState.AddErrors(err)
+		return
+	}
+	if selectType != senseobjdef.SelectTypeListObjectValues {
+		actionState.AddErrors(errors.Errorf("Object id<%s> type<%s> selecttype<%s> datadeftype<%s> must have selecttype<%s> in listboxselect-action",
+			genericObj.GenericId, genericObj.Type, selectType, dataDefType, senseobjdef.SelectTypeListObjectValues))
+		return
+	}
+
+	var selectFunction func(context.Context, string, bool) (bool, error)
+	switch settings.Type {
+	case All:
+		selectFunction = genericObj.SelectListObjectAll
+	case Alternative:
+		selectFunction = genericObj.SelectListObjectAlternative
+	case Excluded:
+		selectFunction = genericObj.SelectListObjectExcluded
+	case Possible:
+		selectFunction = genericObj.SelectListObjectPossible
+	default:
+		actionState.AddErrors(errors.Wrapf(err, "Unknown filter pane selection type ListBoxSelectType<%s>", settings.Type))
+		return
+	}
+
+	selectListObjects := func(ctx context.Context) error {
+		sessionState.LogEntry.LogDebugf("Select in object<%s> h<%d> type<%s>", genericObj.GenericId, genericObj.Handle, genericObj.GenericType)
+
+		success, err := selectFunction(ctx, selectPath, false)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to select in object<%s>", genericObj.GenericId)
+		}
+		if !success {
+			return errors.Errorf("Select in object<%s> unsuccessful", genericObj.GenericId)
+		}
+
+		sessionState.LogEntry.LogDebug(fmt.Sprint("Successful select in", genericObj.GenericId))
+
+		return err
+	}
+
+	if settings.Wrap {
+		beginSelections := func(ctx context.Context) error {
+			return genericObj.BeginSelections(ctx, []string{selectPath})
+		}
+		sessionState.QueueRequest(beginSelections, actionState, false, fmt.Sprintf("Failed to select in %s", genericObj.GenericId))
+		sessionState.Wait(actionState)
+		if actionState.Errors() != nil {
+			return
+		}
```

Comment:
I guess this is copy-pasted, but we have some changes which were implemented since then. 
* There's a actionState.Failed that can be checked instead of Errors()
*  Wait function will return a bool if it was failed or not, so no need to check actionState.Failed in this specific case
* There's now also sessionState.SendRequest which is Sync (we used to only have QueueRequest).

Reply:
I'll just change everything to Sync then. Good point on just looking at the return value if I would keet async.

Reply:
Fixed